### ISSUE TITLE
Modification of the sidebar to add/remove custom volumes/folders

### ISSUE THREAD

**Issue by eljefuri** [elementary/files#689]
Apple's macOS Finder allows customization of the Side Bar, which is crucially helpful for users who work with Files often, and who frequently need access certain folders or volumes.

Selections may be removed by dragging:
https://youtu.be/Hm03IZ7oPME?t=46

Selections may also be added by dragging:
https://youtu.be/Hm03IZ7oPME?t=110

This would be a tremendously helpful addition to Files.

Many thanks to the Files developers for their consideration of this feature.

**Comment by janxkoci** [comment]
Isn't the addition by dragging already supported?

**Comment by jeremypw** [comment]
Yes, drag-dropping a file/folder between bookmarks adds a bookmark at that point.  The UI could probably be improved to make it more obvious - at the moment the only indication that you are dropping in the right place is that the drab emblem changes from "move" to "copy" and then only for local files (otherwise remains "copy").  I think there used to be a line appearing when between bookmarks but this seems to have regressed.

Dragging a bookmark out of the window deletes it.

**Comment by janxkoci** [comment]
Yes now I remembered I once accidentally deleted my bookmark by dragging it onto terminal (I wanted to see if it would open there).

**Comment by janxkoci** [comment]
So if both features are already there, shouldn't we close this? Or do you want to keep it because of the visual clues?

**Comment by jeremypw** [comment]
I'll open another about the visual clue regression.  I notice that the video of creating a bookmark by DnD showed dragging a folder from the desktop which is not applicable to elementary as there is no Desktop folder.  You can only create a bookmark by DnD within Files, not from another program.  I notice that Nautilus does support creating bookmarks by dragging items from a Files window though.

@eljefuri Would you like to update the description of the issue in the light of the above discussion or shall we close it?

**Comment by jeremypw** [comment]
There is already an issue about visual feedback from the sidebar (#699)

**Comment by jeremypw** [comment]
Closing as the desired functions are now present.
